agent/log-drop/log-drop_test.go

```diff
@@ -0,0 +1,37 @@
+package logdrop
+
+import (
+	"context"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestNewLogDrop(t *testing.T) {
+	mockLogger := NewMockLogger(t)
+	mockLogger.On("Log", hclog.Info, "hello", []interface{}{"test", 0}).Return()
+	ld := NewLogDropSink(context.Background(), "test", 10, mockLogger, func(_ Log) {})
+	require.NotNil(t, ld)
+	ld.Accept("test Log", hclog.Info, "hello", "test", 0)
+	retry.Run(t, func(r *retry.R) {
+		mockLogger.AssertNumberOfCalls(r, "Log", 1)
+	})
+
+}
+
+func TestLogDroppedWhenChannelFilled(t *testing.T) {
+	mockLogger := NewMockLogger(t)
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	called := false
+	cancelFunc()
```

Comment:
I think we're canceling the context here to cause the channel to get backed-up. I wonder if a more realistic test would be to have the underlying logger/sink we're wrapping block (e.g. have it receive on a nil channel).